src/test/java/com/jfrog/ide/idea/scan/YarnScannerTest.java

```diff
@@ -0,0 +1,113 @@
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        tempProjectDir = createTempProjectDir("exampleYarnPackage");
+        executorService = ConcurrencyUtil.newSameThreadExecutorService();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        try {
+            executorService.shutdown();
+        } finally {
+            // Ensure that tearDown gets executed even if an exception is thrown
+            super.tearDown();
+        }
+    }
+
+    private String createTempProjectDir(String projectName) throws IOException {
+        // Using a virtual directory allows each test to have its own isolated workspace, preventing interference between tests.
+        String tempProjectDir = getTempDir().createVirtualDir(projectName).toNioPath().toString();
+        FileUtils.copyDirectory(YARN_ROOT.resolve(projectName).toFile(), FileUtils.getFile(tempProjectDir));
+        return tempProjectDir;
+    }
+
+    public void testGetPackageNameToVersionsMap() {
+        YarnScanner yarnScanner = new YarnScanner(getProject(), tempProjectDir, executorService, new GraphScanLogic(new NullLog()));
+        Set<String> packages = new HashSet<>(Arrays.asList("package1:1.0.123", "package2:2.0.9", "package1:1.0.123", "package2:2.1.7"));
+        Map<String, Set<String>> packageNameToVersions = yarnScanner.getPackageNameToVersionsMap(packages);
+
+        assertEquals(2, packageNameToVersions.size());
+        assertTrue(packageNameToVersions.containsKey("package1"));
+        assertTrue(packageNameToVersions.containsKey("package2"));
+        assertFalse(packageNameToVersions.containsKey("package3"));
+
+        assertEquals(new HashSet<>(List.of("1.0.123")), packageNameToVersions.get("package1"));
+        assertEquals(new HashSet<>(Arrays.asList("2.0.9", "2.1.7")), packageNameToVersions.get("package2"));
+    }
+
+    public void testWalkDepTree() throws IOException {
+        GradleProjectImportUtil.linkAndRefreshGradleProject(tempProjectDir, getProject());
+        YarnScanner yarnScanner = new YarnScanner(getProject(), tempProjectDir, executorService, new GraphScanLogic(new NullLog()));
+
+        // Sanity check - make sure the dependency tree is generated.
+        DepTree depTree = yarnScanner.buildTree();
+        assertNotNull(depTree);
+        assertEquals(depTree.getRootId(), "example-yarn-package:1.0.0");
+
+        // Test the walkDepTree method:
+        // Init params
+        String[] vulnerableDependencies = {"lodash:4.16.2", "tough-cookie:2.3.1"};
+        Map<String, DependencyNode> vulnerableDependenciesMap = new HashMap<>();
+        for (String vulnerableDependency : vulnerableDependencies) {
+            DependencyNode dependencyNode = new DependencyNode();
+            dependencyNode.componentId(vulnerableDependency);
+            // Insert a dummy node and remove it to init the children vector.
+            dependencyNode.insert(new DependencyNode(), 0);
```

Comment:
This is a workaround for a bug.
Let's fix the bug by adding a null check in `getSeverity()`:
```java
        Severity severity = Severity.Normal;
        if (children == null) {
            return severity;
        }
```